Firebase.h

```diff
@@ -122,7 +131,7 @@ class FirebasePush : public FirebaseCall {
  public:
   FirebasePush() {}
   FirebasePush(const String& host, const String& auth,
-               const String& path, const String& value, HTTPClient* http = NULL);
+               const String& path, const String& value, FirebaseHttpClient* http = NULL);
```

Comment:
As discussed offline I agree with this but I think it should be a separate PR, it is pretty unrelated to this code and the changes here don't have any user facing effects so they should be safe.